honeybee/properties.py

```diff
@@ -79,4 +49,268 @@ def ToString(self):
+
+    Usage:
+        shade = Shade('Deep Overhang', geometry)
+        shade.properties -> ShadeProperties
+        shade.properties.radiance -> ShadeRadianceProperties
+        shade.properties.energy -> ShadeEnergyProperties
+    """
+
+    def __init__(self, host):
+        """Initialize shade properties.
+
+        Args:
+            host: A Shade object that hosts these properties.
+        """
+        self._host = host
+
+    @property
+    def host(self):
+        """Get the Shade object hosting these properties."""
+        return self._host
+
+    def to_dict(self, include=None):
+        """Convert properties to dictionary.
+
+        Args:
+            include: A list of keys to be included in dictionary.
+                If None all the available keys will be included.
+        """
+        base = {
+            'type': 'ShadeProperties'
+        }
+
+        base = self._add_extension_attr_to_dict(base, include)
+        return base
+
+    def __repr__(self):
+        """Properties representation."""
+        return 'ShadeProperties: {}'.format(self.host.name_original)
+
+
+class ApertureProperties(_Properties):
+    """Honeybee Aperture Properties.
+
+    Aperture properties. This class will be extended by plugins.
+
+    Usage:
+        aperture = Aperture('Window to My Soul', geometry)
+        aperture.properties -> ApertureProperties
+        aperture.properties.radiance -> ApertureRadianceProperties
+        aperture.properties.energy -> ApertureEnergyProperties
+    """
+
+    def __init__(self, host):
+        """Initialize aperture properties.
+
+        Args:
+            host: An Aperture object that hosts these properties.
+            boundary_condition: A honeybee BoundaryCondition object to denote the
+                condition on the other side of the aperture (ie. Oudoors, Surface).
+        """
+        self._host = host
+
+    @property
+    def host(self):
+        """Get the Aperture object hosting these properties."""
+        return self._host
+
+    def to_dict(self, include=None):
+        """Convert properties to dictionary.
+
+        Args:
+            include: A list of keys to be included in dictionary besides
+                boundary_condition. If None all the available keys will be included.
+        """
+        base = {
+            'type': 'ApertureProperties',
+        }
+
+        base = self._add_extension_attr_to_dict(base, include)
+        return base
+
+    def __repr__(self):
+        """Properties representation."""
+        return 'ApertureProperties: {}'.format(self.host.name_original)
+
+
+class DoorProperties(_Properties):
+    """Honeybee Door Properties.
+
+    Door properties. This class will be extended by plugins.
+
+    Usage:
+        door = Door('Front Door', geometry)
+        door.properties -> DoorProperties
+        door.properties.radiance -> DoorRadianceProperties
+        door.properties.energy -> DoorEnergyProperties
+    """
+
+    def __init__(self, host):
```

Comment:
I was considering this but I was also realizing that the `__init__` is really only 2 lines of Python and the majority of the code there is just docstrings.  It also seemed helpful here for the documentation to explain that the host is a `Face that hosts these properties` or a `Model that hosts these properties` since I imagine someone getting confused the first time that they look at how properties work in the API and see that it's not like any other object. I can move it to the base class if you think that I'm overthinking it but I just wanted to explain my reasoning.

Reply:
The reasoning behind the docstring is valid but I would say in case we ever wanted to make a change to \_\_init__ method for these classes reusing the \_\_init__ is the right approach which will save us from editing the code multiple times. I think a generic `Geometry / object that hosts these properties` will do the job for the API.

Reply:
Ok.  I'll move it to the base class then.